tensorstore/kvstore/s3/validate.cc

```diff
@@ -0,0 +1,87 @@
+// Copyright 2020 The TensorStore Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "absl/strings/ascii.h"
+#include "absl/strings/match.h"
+#include "absl/strings/str_split.h"
+
+#include "re2/re2.h"
+
+#include "tensorstore/kvstore/s3/validate.h"
+#include "tensorstore/internal/utf8.h"
+
+namespace tensorstore {
+namespace internal_storage_s3 {
+
+RE2 kIpAddress("^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$");
```

Comment:
LazyRE2 kIpAddress = {"^\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}\\.\\d{1,3}$"};

etc.

Reply:
Neat! I've made these static vars within the `validate.cc` functions.